src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -822,7 +823,11 @@ private void expand(File dir, String[] files, Set<String> cpaths, int version)
                     } else {
                         entryMap.put(name, e);
                     }
-                    expand(f, f.list(), cpaths, version);
+                    String[] dirFiles = f.list();
+                    // Ensure files list is sorted for reproducible jar content
+                    if (dirFiles != null)
```

Comment:
Style issue. I strongly dislike multi-line if statements without braces. I don't know if that style is followed in the rest of the code, but if it seems to fit in I strongly recommend using braces.

Reply:
I agree, we should change this to include braces with the if statement.

Reply:
I agree, I always use {}, I was being consistent with the rest of the expand() method, but I will add {} as I prefer that too, better to set a good example!

Reply:
Feel free to clean those up as well :-)

Reply:
all fixed